### PR TITLE
feat(wix-ui-tpa)dropdown-fixes

### PR DESCRIPTION
**Pull request by urig-wix**
### 🔦 Summary

...

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

**Comment by urig-wix** [comment]
@jonathanadler 
The border width at the examples page is corrupted ("1pxpx").
Any idea where does this style come from?

the Settings Panel seems to be fine though.

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6735

File: packages/wix-ui-tpa/src/components/Dropdown/Dropdown.st.css [hunk 1]
```diff
@@ -97,7 +97,7 @@
     MainBorderDisabledColor: --overridable;
     /*
         The button's border width
-        @default color-1
+        @default 1 px
```
Comment (jonathanadler) [inline packages/wix-ui-tpa/src/components/Dropdown/Dropdown.st.css:100]:
```suggestion
        @default 1px
```

File: packages/wix-ui-tpa/src/components/Dropdown/Dropdown.st.css [hunk 2]
```diff
@@ -215,19 +215,19 @@
...
-    color: value(CurrentButtonTextColor);
-    border-color: value(CurrentButtonBorderColor);
-    border-width: value(CurrentBorderWidth);
-    background: value(CurrentBackgroundColor);
-    font: value(CurrentTextFont);
+    color: value(CurrentButtonTextColor) !important;
+    border-color: value(CurrentButtonBorderColor) !important;
+    border-width: value(CurrentBorderWidth) !important;
+    background: value(CurrentBackgroundColor) !important;
+    font: value(CurrentTextFont) !important;
```
Comment (jonathanadler) [inline packages/wix-ui-tpa/src/components/Dropdown/Dropdown.st.css:230]: So actually, I'm not sure that this part is needed.
What we do need to add is the `!important` in the `path` css rule.
```css
... arrowIcon path {
  fill: ... !important;
  stroke: none !important;
}
```

Comment (jonathanadler) [reply packages/wix-ui-tpa/src/components/Dropdown/Dropdown.st.css:230]: can you check with your app if removing this part still works, or is really needed?

File: packages/wix-ui-tpa/src/components/Dropdown/docs/index.story.tsx [hunk 3]
```diff
@@ -152,7 +152,7 @@ export default {
                     label: 'Button border width',
                     wixParam: 'mainBorderWidth',
                     defaultNumber: 0,
-                    unit: 'px',
+                    unit: '',
```
Comment (jonathanadler) [inline packages/wix-ui-tpa/src/components/Dropdown/docs/index.story.tsx:155]: I think it's ok to leave the `px` here

**Review by jonathanadler — COMMENTED** [review]
So actually I would expect the settings panel to not work.
And in order to fix that you'll need to wrap the variable with the `unit` function here:
https://github.com/wix/wix-style-react/blob/66b67ba2ee9174560442de8a1bfb9caf50450285/packages/wix-ui-tpa/src/components/Dropdown/docs/DropdownExtendedExample.st.css#L12

Also, if you're getting `pxpx` this means somewhere in the `st.css` code we still have a `unit(..., px)` method being used

File: packages/wix-ui-tpa/src/components/Dropdown/Dropdown.st.css (hunk 2 above)
Comment (urig-wix) [reply packages/wix-ui-tpa/src/components/Dropdown/Dropdown.st.css:230]: I think we need both (will add :)
```
... arrowIcon path {
  fill: ... !important;
  stroke: none !important;
}
```
without the `!important` at the `.dropdown .dropdownBase` the placeholder (and text color) are being overrided

Comment (jonathanadler) [reply packages/wix-ui-tpa/src/components/Dropdown/Dropdown.st.css]: 👍

**Comment by jonathanadler** [comment]
@urig-wix 
So just missing the unit in the Settings Panel example, no?

**Comment by urig-wix** [comment]
> @urig-wix
> So just missing the unit in the Settings Panel example, no?

Settings Panel keeps working for me. Interesting.
Which kind of change would you like me to apply?
something like '"applyUnit(value(--mainBorderWidth), px)"'? (<- this change does not work)

**Comment by jonathanadler** [comment]
> > @urig-wix
> > So just missing the unit in the Settings Panel example, no?
> 
> Settings Panel keeps working for me. Interesting.
> Which kind of change would you like me to apply?
> something like '"applyUnit(value(--mainBorderWidth), px)"'? (<- this change does not work)

It should be:
```
'"unit(value(--mainBorderWidth), px)"'
```

**Comment by urig-wix** [comment]
> > > @urig-wix
> > > So just missing the unit in the Settings Panel example, no?
> > 
> > 
> > Settings Panel keeps working for me. Interesting.
> > Which kind of change would you like me to apply?
> > something like '"applyUnit(value(--mainBorderWidth), px)"'? (<- this change does not work)
> 
> It should be:
> 
> ```
> '"unit(value(--mainBorderWidth), px)"'
> ```

'"unit(value(--mainBorderWidth), px)"' does not work

**Comment by jonathanadler** [comment]
> > > > @urig-wix
> > > > So just missing the unit in the Settings Panel example, no?
> > > 
> > > 
> > > Settings Panel keeps working for me. Interesting.
> > > Which kind of change would you like me to apply?
> > > something like '"applyUnit(value(--mainBorderWidth), px)"'? (<- this change does not work)
> > 
> > 
> > It should be:
> > ```
> > '"unit(value(--mainBorderWidth), px)"'
> > ```
> 
> '"unit(value(--mainBorderWidth), px)"' does not work

Wait I got confused sorry 😅  there's no need for the `value` here.
value should never be used with quotation marks.
quotation marks is only for the style-processor functions (`font`, `color`, `opactiy`, `unit`, etc.)
so it should be `'"unit(--mainBorderWidth, px)"'`

**Comment by urig-wix** [comment]
> > > > > @urig-wix
> > > > > So just missing the unit in the Settings Panel example, no?
> > > > 
> > > > 
> > > > Settings Panel keeps working for me. Interesting.
> > > > Which kind of change would you like me to apply?
> > > > something like '"applyUnit(value(--mainBorderWidth), px)"'? (<- this change does not work)
> > > 
> > > 
> > > It should be:
> > > ```
> > > '"unit(value(--mainBorderWidth), px)"'
> > > ```
> > 
> > 
> > '"unit(value(--mainBorderWidth), px)"' does not work
> 
> Wait I got confused sorry 😅 there's no need for the `value` here.
> value should never be used with quotation marks.
> quotation marks is only for the style-processor functions (`font`, `color`, `opactiy`, `unit`, etc.)
> so it should be `'"unit(--mainBorderWidth, px)"'`

Also does not work (to be more accurate - it works after I reset the width by using the scroller at the panel settings. but without changing this value it corrupts the component).

**Comment by jonathanadler** [comment]
> > > > > > @urig-wix
> > > > > > So just missing the unit in the Settings Panel example, no?
> > > > > 
> > > > > 
> > > > > Settings Panel keeps working for me. Interesting.
> > > > > Which kind of change would you like me to apply?
> > > > > something like '"applyUnit(value(--mainBorderWidth), px)"'? (<- this change does not work)
> > > > 
> > > > 
> > > > It should be:
> > > > ```
> > > > '"unit(value(--mainBorderWidth), px)"'
> > > > ```
> > > 
> > > 
> > > '"unit(value(--mainBorderWidth), px)"' does not work
> > 
> > 
> > Wait I got confused sorry 😅 there's no need for the `value` here.
> > value should never be used with quotation marks.
> > quotation marks is only for the style-processor functions (`font`, `color`, `opactiy`, `unit`, etc.)
> > so it should be `'"unit(--mainBorderWidth, px)"'`
> 
> Also does not work (to be more accurate - it works after I reset the width by using the scroller at the panel settings. but without changing this value it corrupts the component).

oh, I know what's the issue here.
You'll need to change it to `'"unit(fallback(--mainBorderWidth, 1), px)"'.

**Comment by urig-wix** [comment]
> > > > > > > @urig-wix
> > > > > > > So just missing the unit in the Settings Panel example, no?
> > > > > > 
> > > > > > 
> > > > > > Settings Panel keeps working for me. Interesting.
> > > > > > Which kind of change would you like me to apply?
> > > > > > something like '"applyUnit(value(--mainBorderWidth), px)"'? (<- this change does not work)
> > > > > 
> > > > > 
> > > > > It should be:
> > > > > ```
> > > > > '"unit(value(--mainBorderWidth), px)"'
> > > > > ```
> > > > 
> > > > 
> > > > '"unit(value(--mainBorderWidth), px)"' does not work
> > > 
> > > 
> > > Wait I got confused sorry 😅 there's no need for the `value` here.
> > > value should never be used with quotation marks.
> > > quotation marks is only for the style-processor functions (`font`, `color`, `opactiy`, `unit`, etc.)
> > > so it should be `'"unit(--mainBorderWidth, px)"'`
> > 
> > 
> > Also does not work (to be more accurate - it works after I reset the width by using the scroller at the panel settings. but without changing this value it corrupts the component).
> 
> oh, I know what's the issue here.
> You'll need to change it to `'"unit(fallback(--mainBorderWidth, 1), px)"'.

you rock you rock you rock

**Pull request merged**